src/tasks

```diff
@@ -1,32 +1,32 @@
 1. Which of the following SOLID principles are violated here?
 (Multiple choice)
-* Single Responsibility Principle (yes)
-* Open-Close principle (yes)
-* Liskov Substitution Principle (no)
-* Inversion of Control (no, tricky, since it's not a SOLID principle :) )
-* Dependency Inversion Principle (yes) 
+* Single Responsibility Principle (yes) [There are multiple reason to modify: print, order, composition]
+* Open-Close principle (yes) [PrintingService is hard-wired, we can't extend it without modifying the NameService]
+* Liskov Substitution Principle (no) [No inheritance]
+* Inversion of Control (no) [Tricky, since it's not a SOLID principle :)]
+* Dependency Inversion Principle (yes) [NameService depends on lower level PrintingService] 
 
 2. Which principle is violated here and why?
 
-* Open-Close Principle, because Square is not Open for extension.
-* Single Responsibility, because Rectangle and Square have the same responsibility
-* Liskov Substitution Principle, because Square can't be used instead Rectangle. (helyes)
-* Interface Segregation Principle, because there is no Interface implemented.
-* Dependency Inversion Principle, because the lowest level Square depends on Rectangle.
+* Open-Close Principle, because Square is not Open for extension. (no)
```

Comment:
Make the format uniform. Above there was answer and reasoning in different brackets. I like that.

Reply:
But reasoning is included to the option here.